Vagrantfile

```diff
@@ -1,33 +1,70 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
+require 'yaml'
+
+ansible_vars = YAML.load_file('provisioning/vars.yml')
+
+app_name = ansible_vars["nodejs_app_name"]
+
+app_directory = ansible_vars["nodejs_app_install_dir"]
+
+# Check for the existence of 'VM_HOST_TCP_PORT' or 'VM_GUEST_TCP_PORT'
+# environment variables. Otherwise if 'nodejs_app_tcp_port' is defined
+# in vars.yml then use that port. Failing that use defaults provided
+# in this file.
+host_tcp_port = ENV["VM_HOST_TCP_PORT"] || ansible_vars["nodejs_app_tcp_port"] || 8080
```

Comment:
We need to actually use the port that the user suggested! There needs to be a single place where the user changes the setting, and all places should respond to that setting - rather than the current situation where there are apparently 4 uncoordinated environment variables. We should also default to a port other than 8080.